packages/salesforce-adapter/src/instance_url.ts

```diff
@@ -0,0 +1,82 @@
+/*
+*                      Copyright 2020 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import {
+  BuiltinTypes, CORE_ANNOTATIONS, ElementIDResolver, ElemID,
+  InstanceElement, isInstanceElement, ObjectType, Element,
+} from '@salto-io/adapter-api'
+import _ from 'lodash'
+import { logger } from '@salto-io/logging'
+import * as constants from './constants'
+import SalesforceClient from './client/client'
+
+const log = logger(module)
+
+// The random suffix is to avoid collisions with salesforce elements
+export const INTERNAL_ACCOUNT_INFO = 'AccountInfo5a2ca8777a7743c3814ec83e3c4f0147'
+
+export const getInstanceUrl = async (elementIDResolver: ElementIDResolver):
+  Promise<URL | undefined> => {
+  const internalAccountInfoElement = await elementIDResolver(new ElemID(constants.SALESFORCE, INTERNAL_ACCOUNT_INFO, 'instance'))
+
+  if (internalAccountInfoElement === undefined || !isInstanceElement(internalAccountInfoElement)) {
+    log.error('Could not found internalAccountInfo element')
+    return undefined
+  }
+  const url = internalAccountInfoElement.value.instanceUrl
+  if (url === undefined) {
+    log.error('internalAccountInfo does not contain instanceUrl')
+    return undefined
+  }
+
+  try {
+    return new URL(url)
+  } catch (e) {
+    log.error(`Failed to parse url: ${e}`)
```

Comment:
1. seems like the error doesn't contain the value so the log message won't be very useful (we won't know what the was value that it failed to parse)
2. logging style - use `%s` instead of constructing the string with a template
```suggestion
    log.error('Failed to parse url %s with error %o', url, e)
```

Reply:
the error does contain the url, e.g., 
```
Failed to parse url: TypeError [ERR_INVALID_URL]: Invalid URL: awdadsawdasd
    at onParseError (internal/url.js:257:9)
    at new URL (internal/url.js:333:5)
    at SalesforceAdapter.fetch (/Users/alon/salto/packages/salesforce-adapter/src/adapter.ts:384:7)
    at Object.timeMethod (/Users/alon/salto/packages/logging/src/internal/logger.ts:80:18)
    at SalesforceAdapter.<anonymous> (/Users/alon/salto/packages/salesforce-adapter/src/adapter.ts:184:11)
    at SalesforceAdapter.wrapped (/Users/alon/salto/packages/lowerdash/src/decorators.ts:40:24)
    at /Users/alon/salto/packages/core/src/core/fetch.ts:243:45
    at Array.map (<anonymous>)
    at fetchAndProcessMergeErrors (/Users/alon/salto/packages/core/src/core/fetch.ts:242:10)
    at Object.exports.fetchChanges (/Users/alon/salto/packages/core/src/core/fetch.ts:337:13)
    at exports.fetch (/Users/alon/salto/packages/core/src/api.ts:218:15)
    at Object.exports.fetchCommand (/Users/alon/salto/packages/cli/src/commands/fetch.ts:123:23)
    at commanderAction (/Users/alon/salto/packages/cli/src/command_builder.ts:178:26)
    at Command.<anonymous> (/Users/alon/salto/packages/cli/src/command_register.ts:121:7)
    at async Promise.all (index 0)
    at Object.exports.default (/Users/alon/salto/packages/cli/src/cli.ts:59:5) {
  [stack]: 'TypeError [ERR_INVALID_URL]: Invalid URL: awdadsawdasd\n' +
    '    at onParseError (internal/url.js:257:9)\n' +
    '    at new URL (internal/url.js:333:5)\n' +
    '    at SalesforceAdapter.fetch (/Users/alon/salto/packages/salesforce-adapter/src/adapter.ts:384:7)\n' +
    '    at Object.timeMethod (/Users/alon/salto/packages/logging/src/internal/logger.ts:80:18)\n' +
    '    at SalesforceAdapter.<anonymous> (/Users/alon/salto/packages/salesforce-adapter/src/adapter.ts:184:11)\n' +
    '    at SalesforceAdapter.wrapped (/Users/alon/salto/packages/lowerdash/src/decorators.ts:40:24)\n' +
    '    at /Users/alon/salto/packages/core/src/core/fetch.ts:243:45\n' +
    '    at Array.map (<anonymous>)\n' +
    '    at fetchAndProcessMergeErrors (/Users/alon/salto/packages/core/src/core/fetch.ts:242:10)\n' +
    '    at Object.exports.fetchChanges (/Users/alon/salto/packages/core/src/core/fetch.ts:337:13)\n' +
    '    at exports.fetch (/Users/alon/salto/packages/core/src/api.ts:218:15)\n' +
    '    at Object.exports.fetchCommand (/Users/alon/salto/packages/cli/src/commands/fetch.ts:123:23)\n' +
    '    at commanderAction (/Users/alon/salto/packages/cli/src/command_builder.ts:178:26)\n' +
    '    at Command.<anonymous> (/Users/alon/salto/packages/cli/src/command_register.ts:121:7)\n' +
    '    at async Promise.all (index 0)\n' +
    '    at Object.exports.default (/Users/alon/salto/packages/cli/src/cli.ts:59:5)',
  input: 'awdadsawdasd',
  [message]: 'Invalid URL: awdadsawdasd',
  code: 'ERR_INVALID_URL'
}
```
(awdadsawdasd is the url)

Changed to %o

Reply:
strange, when I tested it in the chrome console I just got "Invalid URL".
I guess this isn't the same URL class, so never mind